File: Java-Source/Day16.java
//DAY 16 OF HACKERRANK 30 DAY CHALLENGE: EXCEPTIONS - STRING TO INTEGER (JAVA)

import java.util.*;

public class Day16 {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String S = in.next();
        int output;

        try {
            output = Integer.parseInt(S);
            System.out.println(output);
        }
        catch (Exception ex) {
            System.out.println("Bad String");
        }
    }
}

File: CPP-Source/Day16.cpp
//DAY 16 OF HACKERRANK 30 DAY CHALLENGE: EXCEPTIONS - STRING TO INTEGER (C++)

#include <string>
#include <iostream>

using namespace std;


int main(){
    string S;
    cin >> S;
    
    try {
        int output = stoi(S);
        cout << output << endl;
    }
    catch (const invalid_argument& ia) {
        cout << "Bad String" << endl;
    }
    
    return 0;
}